thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/merge/AnomalyMergeExecutor.java

```diff
@@ -1,72 +1,112 @@
 package com.linkedin.thirdeye.anomaly.merge;
 
-import com.linkedin.thirdeye.api.dto.GroupByKey;
-import com.linkedin.thirdeye.api.dto.GroupByRow;
-import com.linkedin.thirdeye.client.pinot.PinotThirdEyeClient;
+import com.linkedin.thirdeye.db.dao.AnomalyFunctionDAO;
 import com.linkedin.thirdeye.db.dao.AnomalyMergedResultDAO;
 import com.linkedin.thirdeye.db.dao.AnomalyResultDAO;
+import com.linkedin.thirdeye.db.entity.AnomalyFunctionSpec;
 import com.linkedin.thirdeye.db.entity.AnomalyMergedResult;
 import com.linkedin.thirdeye.db.entity.AnomalyResult;
+import java.util.ArrayList;
 import java.util.HashMap;
-import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
-import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * finds raw anomalies grouped by a strategy and merges them with an existing (in the same group) or
  * new {@link com.linkedin.thirdeye.db.entity.AnomalyMergedResult}
  */
 public class AnomalyMergeExecutor {
-  AnomalyMergedResultDAO mergedResultDAO;
-  AnomalyResultDAO anomalyResultDAO;
-  AnomalyMergeConfig mergeConfig;
-  PinotThirdEyeClient pinotThirdEyeClient;
+  final AnomalyMergedResultDAO mergedResultDAO;
+  final AnomalyResultDAO anomalyResultDAO;
+  final AnomalyFunctionDAO anomalyFunctionDAO;
+  final AnomalyMergeConfig mergeConfig;
+
+  final static Logger LOG = LoggerFactory.getLogger(AnomalyMergeExecutor.class);
 
   public AnomalyMergeExecutor(AnomalyMergedResultDAO mergedResultDAO,
-      AnomalyResultDAO anomalyResultDAO, AnomalyMergeConfig mergeConfig,
-      PinotThirdEyeClient pinotThirdEyeClient) {
+      AnomalyFunctionDAO anomalyFunctionDAO, AnomalyResultDAO anomalyResultDAO,
+      AnomalyMergeConfig mergeConfig) {
```

Comment:
todo: mergeConfig will be saved with AnomalyFunction. keeping it temporarily here.